nlp/2021_07_07_neural_keyword_extraction/neural_keyword_extraction.ipynb

```diff
@@ -0,0 +1,1520 @@
+        "        \"graph-based, and supervised.\\n\" \n",
+        "        \"This makes pke a great choice to get started with keyword extraction, \"\n",
+        "        \"experiment with different methods and generate baselines to improve upon.\\n\"\n",
+        "        \"But what if the required performance is not met by these classical methods?\\n\"\n",
+        "        \"Newly-developed auspicious extraction methods fall into the category of \"\n",
+        "        \"neural keyword extraction.\\n\"\n",
+        "        \"The methods often utilise sequence-to-sequence models based on recurrent \"\n",
+        "        \"neural networks (RNNs) or Long Short-Term Memory (LSTM). Their objective \"\n",
+        "        \"is to transform a sequence of input words (the given document) into an \"\n",
+        "        \"abstract intermediate representation and generate a sequence of keywords \"\n",
+        "        \"from it.\\n\"\n",
+        "        \"These models do not use words or phrases directly, which enables them to \"\n",
+        "        \"generate unseen keywords as well. This is called keyword generation and \"\n",
+        "        \"combines abstractive as well as extractive keywords.\\n\"\n",
+        "        \"They report impressive performance increases over the classical \"\n",
+        "        \"extraction methods.\\n\"\n",
+        "        \"However, training such models requires a large collection of documents \"\n",
+        "        \"and annotated keywords, since the final training step is usually \"\n",
+        "        \"supervised.\\n\"\n",
+        "        \"In addition, many of the models' repositories are not well maintained \"\n",
+        "        \"which makes it difficult to train them, especially on different \"\n",
+        "        \"languages or domains.\\n\"\n",
+        "        \"This leaves the quesition of how neural keyword extraction can already \"\n",
+        "        \"be used today.\\n\"\n",
+        "        \"Follow the below sections in this notebook to learn how to use two \"\n",
+        "        \"approaches in the neural keyword extraction category and how they \"\n",
+        "        \"compare to classical extraction methods.\"\n",
+        "    ),\n",
+        "]"
+      ],
+      "execution_count": 64,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/",
+          "height": 1000
+        },
+        "id": "RFgnm9mWEQNj",
+        "outputId": "90473f59-16f4-4bb6-9de9-21c7a817a358"
+      },
+      "source": [
+        "# Compare the keywords extracted by the given algorithms\n",
+        "selected_algorithms = ['KeyBERT', 'BART-based', 'TFIDF','YAKE', 'TextRank', 'TopicRank']\n",
+        "\n",
+        "for text in texts:\n",
+        "    compare_keyword_extraction_algorithms(text, selection=selected_algorithms)"
+      ],
+      "execution_count": 66,
+      "outputs": [
```

Comment:
it's a bit strange how the BART-based models are supposed to return the top-3 keywords, but seem to return 2, 2, 6, 2 and 2 keywords respectively, any idea what might cause this?

Reply:
Awesome, thanks 👍